### PR TITLE
Feature/auth routing

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Realm based authentication routing
Depending on the realm defined in the portal (using user@realm) or in the 802.1x connection, the source defined in the realm will be used instead of going though all the sources of the portal profile.
# Impacts

802.1x authentication, portal authentication
# Delete branch after merge

YES
# NEWS file entries
## New Features
- Allowed to specify the authentication source to use depending on the realm

### CODE REVIEW

**Comment by dwlfrth** [comment]
Is it ready or not (https://www.youtube.com/watch?v=bA1PAkKD3Q4) ?

**Comment by julsemaan** [comment]
No no, PR is no ready (https://www.youtube.com/watch?v=MF283pORG2E)

**Comment by CHgsd** [comment]
Works for me ... ( https://www.youtube.com/watch?v=bAuPb16jRjY ) I think.

**Comment by dwlfrth** [comment]
So the "Missing" parts of the description are things of the past ? (https://www.youtube.com/watch?v=0FRIL9PQgac)

**Comment by CHgsd** [comment]
Sorry, I think it is I who missed the timing / drop on this one ( https://www.youtube.com/watch?v=Hl5d6B5zTnY ). Julien, you jump in right behind me. I'll crawl away in shame till I can play with the big kids.

**Comment by dwlfrth** [comment]
Let do this all together shall we ? https://www.youtube.com/watch?v=dKKdJoXF7PI

**Comment by julsemaan** [comment]
Its finally ready :
https://www.youtube.com/watch?v=l72IZMfh1yw

**Comment by dwlfrth** [comment]
Rebase :)

**Comment by dwlfrth** [comment]
Can you rebase this ?

**Comment by julsemaan** [comment]
Rebased

**Comment by fdurand** [comment]
Need do be rebased again

**Comment by julsemaan** [comment]
Rebased

File: lib/pf/role.pm [hunk 1]
```diff
@@ -598,7 +598,7 @@ sub getNodeInfoForAutoReg {
     # under 802.1X EAP, we trust the username provided since it authenticated
     if (defined($args->{'connection_type'}) && (($args->{'connection_type'} & $EAP) == $EAP) && defined($args->{'user_name'})) {
         $logger->debug("EAP connection with a username \"$args->{'user_name'}\". Trying to match rules from authentication sources.");
-        my @sources = ($profile->getInternalSources, $profile->getExclusiveSources );
+        my @sources = get_user_sources($profile, $stripped_user_name, $realm);
```
Comment (fdurand) [inline lib/pf/role.pm]: $stripped_user_name is $args->{'stripped_user_name'} now.
Same for $realm

**Comment by fdurand** [comment]
Tested and works (From the portal, from autoreg)

**Pull request merged**